src/data/reports/glass-data-submission/GLASSDataSubmissionDefaultRepository.ts

```diff
@@ -154,13 +168,77 @@ export class GLASSDataSubmissionDefaultRepository implements GLASSDataSubmission
-            row =>
-                (_.isEmpty(orgUnitIds) || !row.orgUnit ? row : orgUnitIds.includes(row.orgUnit)) &&
-                periods.includes(String(row.period)) &&
-                (completionStatus !== undefined ? row.questionnaireCompleted === completionStatus : row) &&
-                (!submissionStatus ? row : row.status === submissionStatus)
+        const quarterPeriods = _.flatMap(periods, year => quarters.map(quarter => `${year}${quarter}`));
+
+        const filteredRows = rows.filter(row => {
+            return (_.isEmpty(orgUnitIds) || !row.orgUnit ? row : orgUnitIds.includes(row.orgUnit)) &&
+                module !== "EGASP"
+                ? periods.includes(String(row.period))
+                : quarterPeriods.includes(String(row.period)) &&
+                      (completionStatus !== undefined ? row.questionnaireCompleted === completionStatus : row) &&
+                      (!submissionStatus ? row : row.status === submissionStatus);
+        });
+
+        const rowsInPage = _(filteredRows)
+            .orderBy([row => row[sorting.field]], [sorting.direction])
+            .drop((paging.page - 1) * paging.pageSize)
+            .take(paging.pageSize)
+            .value();
+
+        const pager: Pager = {
+            page: paging.page,
+            pageSize: paging.pageSize,
+            pageCount: Math.ceil(filteredRows.length / paging.pageSize),
+            total: filteredRows.length,
+        };
+
+        return {
+            pager,
+            objects: rowsInPage,
+        };
+    }
+
+    async getEAR(
+        options: EARDataSubmissionOptions,
+        namespace: string
+    ): Promise<PaginatedObjects<EARDataSubmissionItem>> {
+        const { paging, sorting, orgUnitIds, from, to, submissionStatus } = options;
+
+        const userOrgUnits = (await this.api.get<{ organisationUnits: Ref[] }>("/me").getData()).organisationUnits.map(
+            ou => ou.id
         );
+        const { organisationUnits } = await this.api
+            .get<any>(
+                "/metadata?organisationUnits:fields=children[children[id,level],id,level],id,level&organisationUnits:filter=level:eq:1"
+            )
+            .getData();
+        const orgUnitsWithChildren = _(userOrgUnits)
+            .flatMap(ou => {
+                const res = flattenNodes(flattenNodes(organisationUnits).filter(res => res.id === ou)).map(
+                    node => node.id
+                );
+                return res;
+            })
+            .uniq()
+            .value();
+
+        const objects = (await this.globalStorageClient.getObject<EARDataSubmissionItem[]>(namespace)) ?? [];
+        const rows = objects.filter(object => orgUnitsWithChildren.includes(object.orgUnit.id));
+
+        const filteredRows = rows
+            .filter(row => {
+                return (
+                    (_.isEmpty(orgUnitIds) || !row.orgUnit ? row : orgUnitIds.includes(row.orgUnit.id)) &&
```

Comment:
Same as above, perhaps it is better to create in variables with descriptive names to understand it better